v01-object-or-null/src/main/java/dev/futa/tutorial/pesel/v01/PeselService.java

```diff
@@ -0,0 +1,112 @@
+package dev.futa.tutorial.pesel.v01;
+
+import dev.futa.tutorial.pesel.Gender;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static dev.futa.tutorial.pesel.Gender.FEMALE;
+import static dev.futa.tutorial.pesel.Gender.MALE;
+import static java.lang.Character.getNumericValue;
+import static java.lang.Integer.parseInt;
+
+public class PeselService {
+
+  private static final byte PESEL_LENGTH = 11;
+  private static final int[] WEIGHTS_PESEL = {1, 3, 7, 9, 1, 3, 7, 9, 1, 3, 1};
+  private static final Pattern PESEL_PATTERN = Pattern.compile("[0-9]{11}");
+
+  private static final Map<Integer, Integer> YEAR_CODES = new HashMap<>();
+
+  static {
+    YEAR_CODES.put(0, 1900);
+    YEAR_CODES.put(20, 2000);
+    YEAR_CODES.put(40, 2100);
+    YEAR_CODES.put(60, 2200);
+    YEAR_CODES.put(80, 1800);
+  }
+
+  /**
+   * Method decodes date od birth and gender from PESEL number
+   *
+   * @param pesel PESEL number to decode data from
+   * @return object contains decoded data and PESEL itself. Will return null if PESEL do not meet
+   *     validation rules or coded date is not valid date.
+   */
+  public PeselInfo decodePesel(final String pesel) {
+
+    if (pesel == null) {
+      return null;
+    }
+
+    if (isPatternValid(pesel) && isChecksumValid(pesel)) {
```

Comment:
Zastanawiam się, czy możliwe jest utworzenie takiego PESELa, który przejdzie walidacje `isPatternValid` oraz `isChecksumValid` natomiast wywali się na `extractBirthDate` (wyjątek wewnątrz, czyli `null` na zewnątrz). Musiałby to być zatem PESEL, który ma np. dzień większy od 31. Przykład
```
PeselInfo peselInfo = peselService.decodePesel("80104001453");
System.out.println(peselInfo);
```
Wynik
```
PeselInfo{pesel='80104001453', birthDate=null, gender=MALE}
```
Zabrakło nam `birthDate` :-(

Reply:
20223003433
19222903431
75123203434
25273203437
25370203433

ale faktycznie masz rację, nie wiem dlaczego nie ma testów z tymi numerami PESEL